--- viewhtmlmail.py	(original)
+++ viewhtmlmail.py	(refactored)
@@ -52,7 +52,7 @@
 
         if part.get_content_subtype() == 'html':
             if html_part:
-                print "Eek, more than one html part!"
+                print("Eek, more than one html part!")
             html_part = part
 
         # Save it to a file in the temp dir.
@@ -68,7 +68,7 @@
         # email message can't be used to overwrite important files.
         # As a first step, guard against ../
         if '../' in filename:
-            print "Eek! Possible security problem in filename", filename
+            print("Eek! Possible security problem in filename", filename)
             continue
 
         filename = os.path.join(tmpdir, filename)
@@ -78,7 +78,7 @@
         # Mailers may use Content-Id or Content-ID (or, presumably, various
         # other capitalizations). So we can't just look it up simply.
         content_id = None
-        for k in part.keys():
+        for k in list(part.keys()):
             if k.lower() == 'content-id':
                 # Remove angle brackets, if present.
                 # part['Content-Id'] is unmutable -- attempts to change it
@@ -97,7 +97,7 @@
                 break     # no need to look at other keys
 
         if not content_id:
-            print filename, "doesn't have a Content-Id, not saving"
+            print(filename, "doesn't have a Content-Id, not saving")
             # print "keys:", part.keys()
 
     # for sf in subfiles:
@@ -118,7 +118,7 @@
     fp.close()
 
     # Now we have the file. Call firefox on it.
-    print "Calling browser " + htmlfile
+    print("Calling browser " + htmlfile)
     os.system("google-chrome " + htmlfile)
 
     # Wait a while to make sure firefox has loads the imgaes, then clean up.
